### DISCUSSION TITLE
Select variables from the draw target

### DISCUSSION THREAD

**Discussion by yizhang-yiz** [wlandau/stantargets#56] [Ideas]
In cmdstanr, for large models it's more efficient to do
```r
CmdStanMCMC_object$draws(variables=c("v1","v2"))
```
than dumping all the variables into the memory. Is there a way to do that in `x_draws_y` target? From the code https://github.com/ropensci/stantargets/blob/7c2e6f4370b2aa285ba9c6b01ad2e70505f1d666/R/tar_stan_mcmc.R#L502 it looks like the target doesn't take that into consideration.

For example, in the following scenario:
```r
fit$draws(c("v1", "v2")) %>% mcmc_dens_overlay()
```
would be more efficient than
```r
tar_read(x_draws_y) %>% posterior::as_draws_df() %>% mcmc_dens_overlay(pars=c("v1", "v2"))
```

**Comment by wlandau** [comment]
The `variables` argument of `tar_stan_mcmc()` does allow you to set specific variables for `x_draws_y`. That line you cited applies to the `x_mcmc_y` target, which saves the whole model object so you can change `variables` without needing to rerun the whole MCMC from scratch. In the short term for you, if you want to avoid saving the whole object, you can instead run `tar_stan_mcmc_rep_draws(variables = "...")` with 1 batch and 1 rep.

Where applicable, I am open to adding entirely new arguments to `tar_stan_mcmc()`, `tar_stan_gq()`, `tar_stan_vb()`, and `tar_stan_mle()` for `variables` and `inc_warmup` to control the big `cmdstanr` object. However, I'm a bit short on time for this. Care to send a pull request?

**Reply by yizhang-yiz** [reply]
Thanks @wlandau. Wouldn't changing `variables` argument in `tar_stan_mcmc` lead to targets reconstruction hence rerun the MCMC? During model exploration I'd like to fit model once, then examine different variables by retrieving them from draws, and generally I wouldn't know before hand which variables to look at. Having to refitting the model is exactly what the package tries to avoid.

**Reply by wlandau** [reply]
The last few commits fix both issues: add new `variables_fit` and `inc_warmup_fit` for the CmdStanFit target separately from the `variables` and `inc_warmup` of the draws target.

**Reply by yizhang-yiz** [reply]
Thanks a lot!
